tests/executors/test_celery_kubernetes_executor.py

```diff
@@ -58,182 +64,104 @@ def test_start(self):
+
+        simple_task_instance = mock.MagicMock()
+        simple_task_instance.queue = test_queue
+
+        cke.queue_command(simple_task_instance, **kwargs)
+
+        if test_queue == KUBERNETES_QUEUE:
+            k8s_queue_cmd.assert_called_once_with(simple_task_instance, *kwarg_values)
+            celery_queue_cmd.assert_not_called()
+        else:
+            celery_queue_cmd.assert_called_once_with(simple_task_instance, *kwarg_values)
+            k8s_queue_cmd.assert_not_called()
+
+    @parameterized.expand(
+        [
+            ('non-kubernetes-queue',),
+            (KUBERNETES_QUEUE,),
+        ]
+    )
+    def test_queue_task_instance(self, test_queue):
+        celery_executor_mock = mock.MagicMock()
+        k8s_executor_mock = mock.MagicMock()
+        cke = CeleryKubernetesExecutor(celery_executor_mock, k8s_executor_mock)
 
-            cke.queue_command(simple_task_instance, command, priority, queue)
+        ti = mock.MagicMock()
+        ti.queue = test_queue
+
+        kwargs = dict(
+            task_instance=ti,
+            mark_success=False,
+            pickle_id=None,
+            ignore_all_deps=False,
+            ignore_depends_on_past=False,
+            ignore_task_deps=False,
+            ignore_ti_state=False,
+            pool=None,
+            cfg_path=None,
+        )
+        kwarg_values = kwargs.values()
+        cke.queue_task_instance(**kwargs)
+        if test_queue == KUBERNETES_QUEUE:
+            k8s_executor_mock.queue_task_instance.assert_called_once_with(*kwarg_values)
+            celery_executor_mock.queue_task_instance.assert_not_called()
+        else:
+            celery_executor_mock.queue_task_instance.assert_called_once_with(*kwarg_values)
+            k8s_executor_mock.queue_task_instance.assert_not_called()
 
-            celery_executor_mock.queue_command.assert_called_once_with(
-                simple_task_instance, command, priority, queue
-            )
-            k8s_executor_mock.queue_command.assert_not_called()
+    @parameterized.expand(
+        [
+            (True, True, True),
+            (False, True, True),
+            (True, False, True),
+            (False, False, False),
+        ]
+    )
+    def test_has_tasks(self, celery_has, k8s_has, cke_has):
+        celery_executor_mock = mock.MagicMock()
+        k8s_executor_mock = mock.MagicMock()
+        cke = CeleryKubernetesExecutor(celery_executor_mock, k8s_executor_mock)
 
-        when_using_k8s_executor()
-        when_using_celery_executor()
+        celery_executor_mock.has_task.return_value = celery_has
+        k8s_executor_mock.has_task.return_value = k8s_has
 
-    def test_queue_task_instance(self):
-        mark_success = False
-        pickle_id = None
-        ignore_all_deps = False
-        ignore_depends_on_past = False
-        ignore_task_deps = False
-        ignore_ti_state = False
-        pool = None
-        cfg_path = None
+        assert cke.has_task(None) == cke_has
```

Comment:
This is different from the original assertion (`assert cke.has_task(ti)`). May you clarify a bit ?

Reply:
so i tend to try to remove things that have no impact on the test
i guess i was thinking that since we're mocking return value, the `ti` object is of no consequence -- the return value would be the same either way.  but i see now that it's worth verifying not only the return behavior and boolean logic, but also just check that the arg is forwarded properly.

HOWEVER, on further inspection, I am noticing that neither CeleryExecutor nor KubernetesExecutor implements `has_task`, so celery.has_task is the same as kubernetes.has_task.  And what this means is that there's really no point in calling k8s.has_task if cel.has_task is false.

this is another case where, if CeleryKubernetesExecutor simply inherited from BaseExecutor, it would avoid some code dupe and complexity.  do you know why it does not inherit from BaseExecutor?

in any case, i'll update the tests to verify call args.

Reply:
ok i see....  there are instance attributes that could differ between the two executors, even though the method is taken from BaseExecutor ----- so it can make sense to check both....  but i still don't understand why this doesn't inherit from BaseExecutor... shouldn't all executors do thisL?